pygeos/test/test_constructive.py

```diff
@@ -306,3 +306,28 @@ def test_reverse_none():
 def test_reverse_invalid_type(geom):
     with pytest.raises(TypeError, match="One of the arguments is of incorrect type"):
         pygeos.reverse(geom)
+
+
+@pytest.mark.parametrize("geometry", all_types)
+def test_clip_by_rect(geometry):
+    actual = pygeos.clip_by_rect([geometry, geometry], 0.0, 0.0, 1.0, 1.0)
+    assert actual.shape == (2,)
+    assert actual[0] is None or isinstance(actual[0], Geometry)
+
+
+def test_clip_by_rect_missing():
+    actual = pygeos.clip_by_rect(None, 0, 0, 1, 1)
+    assert actual is None
+
+
+def test_clip_by_rect_empty():
+    actual = pygeos.clip_by_rect(empty_polygon, 0, 0, 1, 1)
+    assert actual == Geometry("GEOMETRYCOLLECTION EMPTY")
```

Comment:
This seems odd that an empty polygon would be converted to an empty geometry collection

Reply:
This fails outright for empty points:
```python
>>> from pygeos.test.common import empty_point
>>> pygeos.clip_by_rect(empty_point, 0,0,10,10)
...
pygeos.GEOSException: UnsupportedOperationException: getX called on empty Point
```

Whereas intersection does not:
```python
>>> pygeos.intersection(empty_point, pygeos.box(0,0,10,10))
<pygeos.Geometry GEOMETRYCOLLECTION EMPTY>
```

Reply:
The empty points seems a bit strange (and something we can certainly catch), but for the rest this is the behaviour of GEOS. I am not sure to what extent we should start changing this (or otherwise bring it up with them if we think this is strnage behaviour). As you show, `intersection` also returns a geometry collection for empty input:

```
In [12]: pygeos.intersection(pygeos.Geometry("POLYGON EMPTY"), pygeos.box(0,0,10,10))
Out[12]: <shapely.geometry.GeometryCollection GEOMETRYCOLLECTION EMPTY>
```

So that's probably a reason to leave it here as is as well?

Reply:
Yeah, we should probably leave this at the default response from GEOS.